comp/core/workloadmeta/collectors/internal/docker/image_sbom_trivy.go

```diff
@@ -53,7 +52,14 @@ func (c *collector) startSBOMCollection(ctx context.Context) error {
 		workloadmeta.NewFilter(&filterParams),
 	)
 
-	resultChan := make(chan sbom.ScanResult, 2000)
+	scanner := collectors.GetDockerScanner()
+	if scanner == nil {
+		return fmt.Errorf("error retrieving global docker scanner")
+	}
+	resultChan := scanner.Channel()
+	if resultChan == nil {
+		return fmt.Errorf("error retrieving global docker scanner channel")
+	}
```

Comment:
not use in this function

Reply:
Same as above, we use it here:
https://github.com/DataDog/datadog-agent/blob/098c1857e3067db961d613d79a4f9e4d75dbb53f/comp/core/workloadmeta/collectors/internal/docker/image_sbom_trivy.go#L97

Maybe we should have some more generic code between docker and containerd but I leave it for another PR.